packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -353,10 +353,10 @@ export class BaseButton extends React.Component<IBaseButtonProps, IBaseButtonSta
 
     if (menuProps && menuProps.doNotLayer) {
       return (
-        <span style={{ display: 'inline-block' }}>
+        <>
```

Comment:
This seems pretty risky to do in existing widely-used code. @khmakoto or @dzearing (if you weren't the one to suggest this) what do you think?

Reply:
Yeah, I'd say we shouldn't change it in case someone is targeting that specific element in the DOM.

Reply:
Ok, spoke to David offline and the reason for the span wrapper's existence is because this was written before React fragments existed and causes pretty messy bugs since the DOM structure is not inline with normal button. This is indeed a breaking change but it's fundamentally wrong to keep it as is.

My proposal would be to make the change but hold off on merging this PR until we are in the beta phase so that it merges into the compat layer in v8. And since it's going into v8 vs v7 there is less risk in introducing a breaking change.

Reply:
yup let's go with that, thanks all!

Reply:
The fragment change should be unblocked now that 7 has forked off, just be sure to add it in the release notes.